src/common/snippets/src/runtime_configurator.cpp

```diff
@@ -57,18 +44,34 @@ void RuntimeConfigurator::initialization(const lowered::LinearIRCPtr& linear_ir)
     m_latest_shapes.resize(m_io_num);
     m_config->io_data_offsets.resize(m_io_num);
     m_config->tile_rank = linear_ir->get_config().m_loop_depth;
+    m_optimizer.init(linear_ir, m_io_descs, m_in_num);
 }
 
 void RuntimeConfigurator::update(const lowered::LinearIRCPtr& linear_ir) {
+    m_config->master_shape = linear_ir->get_master_shape();
+
+    LoopInfoRuntimeParamsMap initialized_info;
+    auto shapes = extract_shapes();
+    auto layouts = extract_layouts();
+    if (m_optimizer.enabled()) {
+        m_optimizer.optimize(m_config->master_shape, initialized_info, shapes, layouts, m_in_num);
```

Comment:
Why do we need to pass `shapes` and `layouts` to the optimizer explicitly here, when we already passes him `m_io_desc` during the initialization?
The optimizer can extract shapes and layouts internally. It should also be responsible for `tensor_rank` and `data_offsets` update. This would allow to simplify signatures and reduce code connectivity.
As a user, I would expect something like:
```
if (!m_optimizer.optimize(...)) // if m_optimizer.optimize() is true => then the optimizer handled everything correctly
    update_data_offsets(extract_shapes(), extract_layouts());        // if m_optimizer.optimize() is false => optimizer is disabled, and we need to init offsets manually
````

Reply:
I like your suggestion, but currently, `update_data_offsets` is `RuntimeConfigurator`'s protected API, so we can't just call it inside optimizer. I think it will be easier to implement this idea within 148891, when all the needed logic will is performed by lowered passes.

Reply:
>  so we can't just call it inside optimizer
I'm pretty sure we can, because `ParallelWAOptimizer` is an inner class of RuntimeConfigurator: it has access even to the configurators' private members. So the only thing we need to do to call `update_data_offsets` from the optimizer is to pass it a pointer to the configurator: `m_optimizer.optimize(this)`